cobigen/cobigen-core/src/test/java/com/devonfw/cobigen/unittest/config/upgrade/ContextConfigurationUpgraderTest.java

```diff
@@ -34,125 +35,182 @@ public class ContextConfigurationUpgraderTest extends AbstractUnitTest {
-    FileUtils.copyDirectory(new File(templateTestFileRootPath + "/valid-2.0"), cobigen);
-    assertThat(cobigen.toPath().resolve("backup").resolve(pom).toFile()).exists()
-        .hasSameContentAs(cobigen.toPath().resolve(pom).toFile());
+    Path cobigen = this.tempFolder.newFolder(ConfigurationConstants.COBIGEN_CONFIG_FILE).toPath();
+    Path context = cobigen.resolve(ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
+    File sourceTestdata = new File(contextTestFileRootPath + File.separator + currentVersionPath + File.separator
+        + ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
 
-    Path newTemplatesLocation = cobigen.toPath().resolve(ConfigurationConstants.TEMPLATE_SETS_FOLDER)
-        .resolve(ConfigurationConstants.ADAPTED_FOLDER);
-    Path backupContextPath = cobigen.toPath().resolve("backup")
-        .resolve(ConfigurationConstants.CONFIG_PROPERTY_TEMPLATES_PATH)
-        .resolve(ConfigurationConstants.COBIGEN_TEMPLATES).resolve(ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER)
-        .resolve(ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
+    FileUtils.copyDirectory(new File(contextTestFileRootPath + File.separator + currentVersionPath), cobigen.toFile());
 
-    assertThat(backupContextPath.toFile()).exists().hasSameContentAs(context.toFile());
+    ContextConfigurationUpgrader sut = new ContextConfigurationUpgrader();
 
-    for (String s : newTemplatesLocation.toFile().list()) {
-      Path newContextPath = newTemplatesLocation.resolve(s).resolve(ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER);
+    ContextConfigurationVersion version = sut.resolveLatestCompatibleSchemaVersion(context, currentVersion);
+    assertThat(version).as("Source Version").isEqualTo(currentVersion);
 
-      version = sut.resolveLatestCompatibleSchemaVersion(newContextPath);
-      assertThat(version).as("Target version").isEqualTo(ContextConfigurationVersion.getLatest());
+    sut.upgradeConfigurationToLatestVersion(cobigen, BackupPolicy.ENFORCE_BACKUP, targetVersion);
+    assertThat(cobigen.resolve("context.bak.xml").toFile()).exists().hasSameContentAs(sourceTestdata);
 
-      newContextPath = newContextPath.resolve(ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
-      XMLUnit.setIgnoreWhitespace(true);
-      new XMLTestCase() {
-      }.assertXMLEqual(new FileReader(contextTestFileRootPath + "/valid-" + ContextConfigurationVersion.getLatest()
-          + "/" + s + "/" + ConfigurationConstants.CONTEXT_CONFIG_FILENAME), new FileReader(newContextPath.toFile()));
+    version = sut.resolveLatestCompatibleSchemaVersion(cobigen, targetVersion);
+    assertThat(version).as("Target version").isEqualTo(targetVersion);
 
-    }
+    XMLUnit.setIgnoreWhitespace(true);
+    new XMLTestCase() {
+    }.assertXMLEqual(new FileReader(contextTestFileRootPath + File.separator + targetVersionPath + File.separator
+        + ConfigurationConstants.CONTEXT_CONFIG_FILENAME), new FileReader(context.toFile()));
```

Comment:
two filereader not in try-with-resource => file lock issue / memory leak - please check all test cases